### DISCUSSION TITLE
prometheus data source variables not available when using mixed datasoures

### DISCUSSION THREAD

**Discussion by RichardWright** [grafana/grafana#42594] [Dashboards, Panels & Visualizations]
I've created a table using three seperate prometheus data sources for three seperate queries (dev, staging prod as examples). I'd like to use variables for the data sources, when the data source dropdowns on the queries don't feature any of my variables when I look. Is this a bug?

here you can see the name of the variable, but changing the dropdown it goes away
<img width="545" alt="Screenshot 2021-12-01 at 14 52 35" src="https://user-images.githubusercontent.com/881815/144258055-e46012de-6e01-407b-ad7f-b4377e984f9f.png">

<img width="545" alt="Screenshot 2021-12-01 at 14 53 21" src="https://user-images.githubusercontent.com/881815/144258027-7f882b65-0db5-4b66-acdf-463092c1e541.png">

**Comment by xuhrc** [comment]
I have the same question 
https://github.com/grafana/grafana/discussions/42913

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
